### PR TITLE
Improve (plugin/trino-iceberg, minio-and-avro) performance

### PR DESCRIPTION
**Pull request by homar**
## Description
Checking ci workflows executions showed that `testStatsBasedRepartitionDataOnCtas`, `testStatsBasedRepartitionDataOnInsert` and `testMergeLarge` are the slowest test in the suite. This PR's goal is to try to decrease their execution time 

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java [hunk 1]
```diff
@@ -4574,18 +4574,7 @@ public Object[][] repartitioningDataProvider()
     }
 
     @Test
-    public void testStatsBasedRepartitionDataOnCtas()
```
Comment (alexjo2144) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:4577]: You're saving yourself a little bit of CTAS time by not having to recreate the table but you're loosing some parallelism by not being able to run both tests at the same time.

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 2]
```diff
@@ -5646,16 +5646,14 @@ public void testMergeLarge()
 
         assertUpdate(mergeSql, 1_000_000);
 
-        // verify deleted rows
-        assertQuery("SELECT count(*) FROM " + tableName + " WHERE mod(orderkey, 3) = 1", "SELECT 0");
-
+        //there is no need to verify deletes as number of mod(orderkey, 3) = 2 and mod(orderkey, 3) equals number of rows in table
```
Comment (alexjo2144) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5649]: I don't really understand this comment

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 3]
```diff
@@ -5646,16 +5646,14 @@ public void testMergeLarge()
...
         // verify untouched rows
         assertThat(query("SELECT count(*), cast(sum(totalprice) AS decimal(18,2)) FROM " + tableName + " WHERE mod(orderkey, 3) = 2"))
-                .matches("SELECT count(*), cast(sum(totalprice) AS decimal(18,2)) FROM tpch.sf1.orders WHERE mod(orderkey, 3) = 2");
+                .matches("SELECT BIGINT '500000', cast('75683551966.03' AS DECIMAL(18, 2))");
```
Comment (alexjo2144) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:5652]: How about changing the query to do `ceiling(sum(totalprice))`?

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java (hunk 1 above)
Comment (homar) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:4577]: right, good point

**Pull request closed without merging**